### PR TITLE
Simple perf replication test using docker containers

### PR DESCRIPTION
**Pull request by ncabatoff**
This one's a bit more experimental, in that the ReplicationSet approach is different from how we're grouping clusters in our tests at present.  I'm trying to steer us in the direction of allowing arbitrary collections of clusters, and of them having names that aren't based on their intended initial roles.

Although this relies on an Enterprise feature (replication), most of this code will live in the sdk.  The actual tests will live in ent only though, since they rely on ent binaries that won't be available in the OSS CI.  For now I'm including an example test for reference.  The test can actually be run in a local checkout of OSS, provided you set VAULT_BINARY to point to an enterprise binary.

One thing that I probably still want to improve is a bit of UX.  This log is pretty convenient:
```
t.Logf("Cluster %s started: VAULT_TOKEN=%s VAULT_CACERT=/vault/config/ca.pem", name, cluster.RootToken())
```

since otherwise, if you want to debug by getting a terminal in a container, you might know how to set VAULT_CACERT so that you can at least use unauth endpoints, but without a token you won't be able do more than that.  I want this for non-replication clusters too.

Also, right now the nodeids used by clusters aren't tied to each cluster, e.g. core-0 is named core-0 in every cluster.  Log lines are prefixed with the cluster name, but there are still cases where I think it might be less confusing if a short version of the cluster name were prepended to each nodeid.  I may or may not get to that in this PR, this is mostly a reminder to myself for now that this would be desirable.

### CODE REVIEW

File: sdk/helper/testcluster/docker/replication.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+					// If you want the test to run faster locally, you could
+					// uncomment this performance_multiplier change.
+					//StorageOptions: map[string]string{
+					//	"performance_multiplier": "1",
+					//},
+				},
+			},
+			CA: r.CA,
+		})
+		t.Logf("Cluster %s started: VAULT_TOKEN=%s VAULT_CACERT=/vault/config/ca.pem", name, cluster.RootToken())
```
Comment (hghaf099) [inline sdk/helper/testcluster/docker/replication.go:47]: This is very helpful!

**Review by hghaf099 — APPROVED** [review]
Looks good to me! I just left a couple of nit comments.

File: sdk/helper/testcluster/docker/replication.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+				VaultNodeConfig: &testcluster.VaultNodeConfig{
+					LogLevel: "TRACE",
+					// If you want the test to run faster locally, you could
+					// uncomment this performance_multiplier change.
+					//StorageOptions: map[string]string{
+					//	"performance_multiplier": "1",
+					//},
+				},
+			},
+			CA: r.CA,
```
[the remark this answers is not in the document]
Comment (ncabatoff) [reply sdk/helper/testcluster/docker/replication.go:45]: The license is handled in NewTestDockerCluster now.

File: sdk/helper/testcluster/docker/replication.go (hunk 1 above)
Comment (ncabatoff) [reply sdk/helper/testcluster/docker/replication.go:47]: Yeah, but we need to also add something like this for after we generate a new token as part of enabling replication on secondary I think.  Might've seen something like that in your PR?

File: sdk/helper/testcluster/replication.go [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+		return err
+	}
+	return WaitForActiveNodeAndPerfStandbys(ctx, pri)
+}
+
+func WaitForPerfReplicationState(ctx context.Context, cluster VaultCluster, state consts.ReplicationState) error {
+	client := cluster.Nodes()[0].APIClient()
+	var health *api.HealthResponse
+	var err error
+	for ctx.Err() == nil {
```
[the remark this answers is not in the document]
Comment (ncabatoff) [reply sdk/helper/testcluster/replication.go:52]: The idea is that the caller can choose to create a context for however long they want to wait.  I agree it likely makes sense to add some timeouts at some level, I just haven't figured out what sensible values for CI are yet. Maybe something like a 1m timeout in TestStandardPerfReplication_Docker and NewReplicationSetDocker (for the initial primary cluster creation) makes sense?

[the remark this answers is not in the document]
Comment (hghaf099) [reply sdk/helper/testcluster/replication.go:52]: Yeah, I think 1 minute is a good initial timeout. We may adjust it later if needed.

File: sdk/helper/testcluster/docker/replication.go (hunk 1 above)
Comment (hghaf099) [reply sdk/helper/testcluster/docker/replication.go:47]: Good point! I have not done that. I will add something. Though I have not passed in `t` to those functions. I will think of something.

Comment (ncabatoff) [reply sdk/helper/testcluster/docker/replication.go]: I decided to switch to using the cluster logger instead of t, see latest commit.

**Pull request merged**